types/three/src/renderers/shaders/ShaderLib.d.ts

```diff
@@ -1,9 +1,8 @@
+import { WebGLProgramParameters } from '../webgl/WebGLProgramParameters.js';
 import { IUniform } from './UniformsLib.js';
 
-export interface Shader {
+export interface Shader extends WebGLProgramParameters {
```

Comment:
Should I rename `Shader` to `WebGLProgramParametersWithUniforms`? I think `Shader` works too as a parameter type.

Reply:
I was actually thinking we'd keep `Shader` as-is, since it's really just intended to describe the type of shaders in `ShaderLib` Maybe it should be renamed to `ShaderLibShader` to make that more clear? I was thinking we'd export a `WebGLProgramParametersWithUniforms` interface from `WebGLRenderer`.

Reply:
I've exported `WebGLProgramParametersWithUniforms` from `WebGLPrograms` since it's used there as a param type for `acquireProgram`. Is there a specific reason why it should be in `WebGLRenderer`?

Reply:
No, that should be fine, thanks.